src/Screencast/main.vala

```diff
@@ -0,0 +1,81 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1 --pkg libportal-gtk4 --pkg gstreamer-1.0
+
+private Gtk.Picture output;
+
+async void start_screencast_session (Xdp.Portal portal, Xdp.Parent parent) {
+    try {
+        Xdp.Session session = yield portal.create_screencast_session (Xdp.OutputType.MONITOR,
+            Xdp.ScreencastFlags.NONE,
+            Xdp.CursorMode.EMBEDDED,
+            Xdp.PersistMode.TRANSIENT,
+            null,
+            null);
+
+        if (session == null) {
+            message ("Permission denied");
+            return;
+        }
+
+        var success = yield session.start (parent, null);
+
+        if (!success) {
+            message ("Could not start session");
+            return;
+        }
+        var pw_remote = session.open_pipewire_remote ();
+        var pipeline = new Gst.Pipeline ("");
+        var source = Gst.ElementFactory.make ("pipewiresrc", "source");
```

Comment:
All elements created using `Gst.ElementFactory.make` should be `Gst.Element` instead of `var`, to make the type returned more clear.